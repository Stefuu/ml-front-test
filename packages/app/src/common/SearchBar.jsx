import React from 'react'
import Box from '../components/elements/Box'
import RetinaImage from '../components/behaviour/RetinaImage'
import Logo from '../assets/Logo_ML.png'
import RetinaLogo from '../assets/Logo_ML@2x.png'

const SearchBar = () => (
  <Box>
    <Box>
      <RetinaImage height='36px' regular={Logo} retina={RetinaLogo} alt='' />
    </Box>
  </Box>
)

export default SearchBar
