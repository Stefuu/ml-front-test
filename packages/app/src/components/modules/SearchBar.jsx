
import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import { rem } from 'polished'
import styled from 'styled-components'
import { theme } from 'styled-tools'
import Container from '../modules/container'
import Flex from '../elements/Flex'
import Box from '../elements/Box'
import InputText from '../inputs/InputText'
import SearchButton from '../inputs/SearchButton'
import RetinaImage from '../behaviour/RetinaImage'
import LogoImageRegular from '../../assets/Logo_ML.png'
import LogoImageRetina from '../../assets/Logo_ML@2x.png'
import SearchImageRegular from '../../assets/ic_Search.png'
import SearchImageRetina from '../../assets/ic_Search@2x.png'

const StyledBox = styled(Box)`
  padding: 10px 0px 10px 10px;
  background-color: ${theme('colors.secondary')};
`

const SearchBar = (props) => (
  <StyledBox>
    <Container>
      <Flex>
        <Box as={Link} to='/' mr={rem(27)}>
          <RetinaImage
            width={rem(53)}
            height={rem(36)}
            regular={LogoImageRegular}
            retina={LogoImageRetina}
            alt='Logo' />
        </Box>
        <Flex
          onSubmit={props.onSubmit}
          mr={rem(5)}
          as={'form'}
          width='100%'>
          <InputText
            fluid
            onChange={props.onChange}
            value={props.searchTerm}
            type='text'
            placeholder='Nunca dejes de buscar' />
          <SearchButton onClick={props.onSubmit}>
            <RetinaImage
              width={rem(20)}
              height={rem(18)}
              regular={SearchImageRegular}
              retina={SearchImageRetina}
              alt='Logo' />
          </SearchButton>
        </Flex>
      </Flex>
    </Container>
  </StyledBox>
)

SearchBar.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  searchTerm: PropTypes.string
}

export default SearchBar
