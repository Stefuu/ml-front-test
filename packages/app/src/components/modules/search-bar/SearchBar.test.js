import React from 'react'
import { render, fireEvent } from 'react-testing-library'
import { BrowserRouter } from 'react-router-dom'
import SearchBar from './SearchBar'

describe('SearchBar', () => {
  it('Should render [snapshot]', async () => {
    const { container } = render(
      <BrowserRouter>
        <SearchBar />
      </BrowserRouter>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should be able to type the search term', async () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <SearchBar />
      </BrowserRouter>
    )

    let input
    input = getByTestId('input-search')
    expect(input.value).toBe('')
    fireEvent.change(input, { target: { value: 'TEST VALUE' } })
    expect(input.value).toBe('TEST VALUE')
  })
})
