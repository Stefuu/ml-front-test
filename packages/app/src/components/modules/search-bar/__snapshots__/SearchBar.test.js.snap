// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Should render [snapshot] 1`] = `
<div
  class="sc-bdVaJa sc-ifAKCX bJlpkh"
>
  <div
    class="sc-bdVaJa fEwoCY"
  >
    <div
      class="sc-bdVaJa cyMNyE"
    >
      <a
        class="sc-bdVaJa cpotsO"
        href="/"
        mr="1.6875rem"
      >
        <div
          alt="Logo"
          class="sc-bxivhb jLbKpf"
          height="2.25rem"
          width="3.3125rem"
        />
      </a>
      <form
        class="sc-bdVaJa gGeGEh"
        display="flex"
        width="100%"
      >
        <input
          class="sc-bwzfXH OeBeg"
          data-testid="input-search"
          placeholder="Nunca dejes de buscar"
          type="text"
          value=""
        />
        <button
          class="sc-htpNat jQvakz"
          data-testid="search-button"
        >
          <div
            alt="Logo"
            class="sc-bxivhb hGGnaW"
            height="1.125rem"
            width="1.25rem"
          />
        </button>
      </form>
    </div>
  </div>
</div>
`;
